server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -1146,6 +1181,7 @@ public Metadata apply(Metadata part) {
             builder.indices(indices.apply(part.indices));
             builder.templates(templates.apply(part.templates));
             builder.customs(customs.apply(part.customs));
+            builder.put(operatorMetadata.apply(part.operatorMetadata));
```

Comment:
Does the jdk map diff return an immutable/unmodifiable map? We could either wrap or copy this here to make it immutable, or even better open a separate issue to make the jdk map diff's apply return an immutable map (I've had the need for this in recent hppc PRs)